src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -561,11 +549,7 @@ public async Task<IActionResult> Enable(string id)
                 return Forbid();
             }
 
-            var shellSettings = _shellHost.GetAllSettings()
-                .Where(x => String.Equals(x.Name, id, StringComparison.OrdinalIgnoreCase))
-                .FirstOrDefault();
-
-            if (shellSettings == null)
+            if (!_shellHost.TryGetSettings(id, out var shellSettings))
```

Comment:
Also look at `TenantValidator.cs` for another place to use `_shellHost.TryGetSettings()`.

https://github.com/OrchardCMS/OrchardCore/blob/af69cdb09a2959a76583f6538f20b9ae9cd185ac/src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs#L80-L85

Reply:
Yes, in the TeantValiadator we need all settings anyway to check for existing prefixes

Reply:
Okay as you want